src/hooks/projections/useUpdateDraftedProjection.ts

```diff
@@ -0,0 +1,103 @@
+import type { Schema } from 'src/types';
+import type { Projections } from 'src/types/schemaModels';
+
+import { useCallback } from 'react';
+
+import { omit } from 'lodash';
+
+import { modifyDraftSpec } from 'src/api/draftSpecs';
+import { useBindingsEditorStore_collectionData } from 'src/components/editor/Bindings/Store/hooks';
+import { useEditorStore_persistedDraftId } from 'src/components/editor/Store/hooks';
+
+const getExistingPartition = (spec: Schema, location: string) => {
+    const existingProjection = spec?.projections
+        ? Object.entries(spec.projections as Projections).find(
+              ([_projectedField, projectedMetadata]) =>
+                  typeof projectedMetadata === 'string'
+                      ? projectedMetadata === location
+                      : projectedMetadata.location === location
+          )
+        : undefined;
+
+    if (!existingProjection || typeof existingProjection[1] === 'string') {
+        return undefined;
+    }
+
+    return existingProjection[1].partition;
+};
+
+export const useUpdateDraftedProjection = () => {
+    const draftId = useEditorStore_persistedDraftId();
+    const collectionData = useBindingsEditorStore_collectionData();
+
+    const removeSingleProjection = useCallback(
+        async (collection: string, field: string) => {
+            if (!draftId || !collectionData) {
+                return Promise.reject(
+                    'draft ID or stored draft spec not found'
+                );
+            }
+
+            const spec: Schema = collectionData.spec;
+
+            if (spec?.projections) {
+                spec.projections = omit(spec.projections, field);
```

Comment:
Support for `omit` is gonna go away. This object _should_ normally be small so the perf hit isn't bad. Probably want to start using `delete` directly.